specification/attributes/key_value_format.md

```diff
@@ -0,0 +1,34 @@
+# Key-Value Format
+
+This is the specified format for providing data in the form of key-value pairs. In addition, this can support keyword data which has a key but not value. This format is used for data such as tags, labels, and resource metadata. 
+
+## Attribute ID
+
+KeyValueFormat
+
+## Attribute Name
+
+Key-Value Format
+
+## Description
+
+Rules and formatting requirements for columns appearing in billing data which convey key-value pairs and/or keyword attributes.
+
+## Requirements
+
+* Data MUST be provided as a string in a format consistent with ECMA-404, as a single object.
+  * When applied to the key-value pair, "string" refers to the key and "value" refers to the value.
+* Keys MUST be unique within an object.
+  * In the case of key collisions, multiple sets of key-value pairs can be provided by prefixing tag keys to disambiguate sources.
+* Keys SHOULD be case-insensitive.
```

Comment:
I have concerns about this.....many data systems that deal in JSON, keys are case sensitive.    Databse like Redshift, Mongo, MySQL all support JSON fields and are all case sensitive. The JSON spec itself is case sensitive on keys (in that you can do it), therefore python, java, R, etc....they use case sesntive keys when dealing with JSON. I have concerns if we are adding this requirement...it makes our JSON special...I don't think it should be.

Reply:
Absolutely fair and up for discussion. The JSON spec explicitly does not have these restrictions but does allow for specifications using JSON to apply these restrictions.

Per ECMA 404, section 6:
> The JSON syntax does not impose any restrictions on the strings used as names, does not require that name strings be unique, and does not assign any significance to the ordering of name/value pairs. These are all semantic considerations that may be defined by JSON processors or in specifications defining specific uses of JSON for data interchange.

Reply:
I think this is only an issue for the usability tags. I'd move it there. Fwiw, it's highly unlikely that any vendor would use 2 attributes with the same name, but different casing. (I'm going to regret writing this, aren't I? 🤦‍♂️)